util/solver/parser/load_spec.py

```
@@ -0,0 +1,283 @@
+    spec_path, class_name = split_problem_ref(problem_ref)
+    module = _load_module(spec_path)
+    if not hasattr(module, class_name):
+        raise SpecLoadError(f"{spec_path} does not define class {class_name}")
+    problem_cls = getattr(module, class_name)
+    if not isinstance(problem_cls, type) or not issubclass(problem_cls, SolveSpec):
+        raise SpecLoadError(f"{class_name} is not a SolveSpec subclass")
+    return spec_path, class_name, problem_cls
+
+
+def _load_module(spec_path: Path):
+    module_name = f"solver_spec_{sha1(str(spec_path).encode('utf-8')).hexdigest()}"
```

Ruff (0.15.21)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)